### DISCUSSION TITLE
How can I calculate metrics over all of the sequences

### DISCUSSION THREAD

**Discussion by icedumpy** [neuralhydrology/neuralhydrology#104] [Q&A]
I trained a model `(cudalstm)` that forecast 168 hours ahead and I want to calculate the metrics with all of the 168 hours prediction.

However, when I ran the code, it shown that 
`Metrics for 1H are calculated over last 1 elements only. Ignoring 167 predictions per sequence.`

Does that mean it only calculate 1 hour ahead?
If so, how can I measure all of the sequence. Thank you

**Comment by kratzert** [comment]
Sorry for the late reply, I must have missed your question:

What happens here is that each sample in your training/validation data is shifted by one hour. So two consecutive samples have an overlap of 167 timesteps in your case. What we do during validation is simply to stack the last output (last timestep) of each sample into one timeseries and compute this NSE. If you want to compute the NSE over the 168 timesteps of each sample and then average the NSE over each of the sequences, this would be something you need to do with the saved data in your run folder.
